contractsV5/KyberTradeLogic.sol

```diff
@@ -212,49 +211,51 @@ contract KyberTradeLogic is KyberHintParser, IKyberTradeLogic, PermissionGroups
-        if (hint.length == 0 || hint.length == 4) {
-            tradeData.tokenToEth.isFeePaying = new bool[](1);
-            tradeData.tokenToEth.splitValuesBps = new uint[](1);
-            tradeData.tokenToEth.rates = new uint[](1);
-            tradeData.ethToToken.isFeePaying = new bool[](1);
-            tradeData.ethToToken.splitValuesBps = new uint[](1);
-            tradeData.ethToToken.rates = new uint[](1);
+        // relevant arrays will be initialised when storing data
+        if (hint.length == 0 || hint.length == 4) return;
+
+        if (src == ETH_TOKEN_ADDRESS) {
+            (
+                tradeData.ethToToken.tradeType,
+                tradeData.ethToToken.addresses,
+                tradeData.ethToToken.splitValuesBps
+            ) = parseHintE2T(hint);
+        } else if (dest == ETH_TOKEN_ADDRESS) {
+            (
+                tradeData.tokenToEth.tradeType,
+                tradeData.tokenToEth.addresses,
+                tradeData.tokenToEth.splitValuesBps
+            ) = parseHintT2E(hint);
         } else {
-            if (src == ETH_TOKEN_ADDRESS) {
-                (
-                    tradeData.ethToToken.tradeType,
-                    tradeData.ethToToken.addresses,
-                    tradeData.ethToToken.splitValuesBps,
-                    failingIndex
-                ) = parseE2tHint(hint);
-            } else if (dest == ETH_TOKEN_ADDRESS) {
-                (
-                    tradeData.tokenToEth.tradeType,
-                    tradeData.tokenToEth.addresses,
-                    tradeData.tokenToEth.splitValuesBps,
-                    failingIndex
-                ) = parseT2eHint(hint);
-            } else {
-                (
-                    tradeData.tokenToEth.tradeType,
-                    tradeData.tokenToEth.addresses,
-                    tradeData.tokenToEth.splitValuesBps,
-                    tradeData.ethToToken.tradeType,
-                    tradeData.ethToToken.addresses,
-                    tradeData.ethToToken.splitValuesBps,
-                    failingIndex
-                ) = parseT2tHint(hint);
-            }
+            (
+                tradeData.tokenToEth.tradeType,
+                tradeData.tokenToEth.addresses,
+                tradeData.tokenToEth.splitValuesBps,
+                tradeData.ethToToken.tradeType,
+                tradeData.ethToToken.addresses,
+                tradeData.ethToToken.splitValuesBps
+            ) = parseHintT2T(hint);
+        }
 
-            require(failingIndex > 0);
+        //init rates and feePaying arrays if split hint
+        if (tradeData.tokenToEth.tradeType == TradeType.Split) {
+            tradeData.tokenToEth.rates = new uint[](tradeData.tokenToEth.addresses.length);
+            tradeData.tokenToEth.isFeePaying = new bool[](tradeData.tokenToEth.addresses.length);
         }
+
+        if (tradeData.ethToToken.tradeType == TradeType.Split) {
+            tradeData.ethToToken.rates = new uint[](1);
```

Comment:
seems to me numbers should be as above.
so if trade type is split.
should be split values and fee paying values same as number of reserves.